tooling/nargo_cli/src/cli/mod.rs

```diff
@@ -223,31 +222,32 @@ fn lock_workspace(
         }
     }
 
-    let mut locks = Vec::new();
-    for pkg in workspace.into_iter() {
-        let toml_path = get_package_manifest(&pkg.root_dir)?;
-        let path_display = toml_path.display();
-
-        let file = File::open(&toml_path)
-            .unwrap_or_else(|e| panic!("Expected {path_display} to exist: {e}"));
-
-        if exclusive {
-            if fs2::FileExt::try_lock_exclusive(&file).is_err() {
-                eprintln!("Waiting for lock on {path_display}...");
-            }
-            fs2::FileExt::lock_exclusive(&file)
-                .unwrap_or_else(|e| panic!("Failed to lock {path_display}: {e}"));
-        } else {
-            if fs2::FileExt::try_lock_shared(&file).is_err() {
-                eprintln!("Waiting for lock on {path_display}...",);
-            }
-            fs2::FileExt::lock_shared(&file)
-                .unwrap_or_else(|e| panic!("Failed to lock {path_display}: {e}"));
-        }
+    let lock_path = target_dir.join(".nargo-lock");
```

Comment:
I would suggest locking on a package specific lock file in the output directory, to allow the compilations in aztec-packages to go concurrently without all waiting on this single `.nargo-lock` to be free.